test/components/bugzilla.test.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import BugzillaQueryUrl from '../../src/components/BugzillaUrl';
+
+describe('<BugzillaQueryUrl />', () => {
+  it('renders correctly', () => {
+    const tree = renderer
+      .create(
+        <BugzillaQueryUrl
+          bugCount={1}
+          url="http://github.com"
+          text="test"
+          classes={{
+            padding: 'X',
+            text: 'Y',
+            margin: 'Z',
+          }}
+        />
+      )
+      .toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders correctly without bugs', () => {
+    const tree = renderer
+      .create(
+        <BugzillaQueryUrl
+          url="http://github.com"
+          text="test"
+          classes={{
+            padding: 'X',
+            text: 'Y',
+            margin: 'Z',
+          }}
+        />
+      )
+      .toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+});
```

Comment:
These two tests cover just one line of code.  Remove the one that is providing no net-new-coverage.

The remaining test will cover just one line.  I looked through the code and the `BugzillaQueryUrl` is used by only by `BugzillaUrlContainer`, which is used only by `src/views/Android/index.jsx`  Can a higher level test provide more coverage?

Reply:
I think thats ok to leave both. Because here we are testing the component it self. Even though another test of a higher component tests it, the correct is to each component have a test. What do you think?

Reply:
@BrunoDM2943, 

There is this the general issue of scattered implementation: from [this comment](https://github.com/mozilla-frontend-infra/firefox-health-dashboard/issues/288#issue-406416189) I mention this principal:

> Only use indirection if more than one component/method/object must reference it.

This component should be in the same file as `BugzillaUrlContainer`, and the test(s) should be on that higher component instead.   Even better: the `BugzillaUrlContainer` (allong with its unique dependencies) should be in` Android/index.jsx`, because it is the only place it is used; the test should be on the `Android` component.  If you inline this chain of code, you will find that some code paths are impossible, and the lines of code collapses and zero new tests are needed. 

This is one of many instances where this codebase exposes a public entity (`BugzillaQueryUrl`), which really should be private so it does not contribute to the perceived complexity of the code.

Some functions do not need tests: 
* functions that are provably correct - so simple that they are obvious or a single pass through the debugger can show error; in this case a visual inspection is good enough.
* private functions - functions that are part of a larger function

Adding tests to these types of functions makes the the application more difficult to change; they are testing "how" the function is implemented rather than "what" needs to be done. 

I thank you for the effort providing tests, but you stumbled upon a function that really should be cleaned up.